File: server/src/main/java/au/com/codeka/warworlds/server/store/StoreException.java
package au.com.codeka.warworlds.server.store;

import java.sql.SQLException;

/**
 * Exception that is thrown from the some of the store classes when an unrecoverable error occurs.
 */
public class StoreException extends Exception {
  /** Constructs a new {@link StoreException} that wraps the given {@link SQLException}. */
  public StoreException(SQLException e) {
    super(e.getMessage(), e);
  }

  /** Constructs a new {@link StoreException} with the given exception message. */
  public StoreException(String msg) {
    super(msg);
  }

  /** Constructs a new {@link StoreException} with the given exception. */
  public StoreException(Exception e) {
    super(e);
  }
}

File: server/src/main/java/au/com/codeka/warworlds/server/store/StoreException.kt
package au.com.codeka.warworlds.server.store

import java.sql.SQLException

/**
 * Exception that is thrown from the some of the store classes when an unrecoverable error occurs.
 */
class StoreException : Exception {
  /** Constructs a new [StoreException] that wraps the given [SQLException].  */
  constructor(e: SQLException) : super(e.message, e) {}

  /** Constructs a new [StoreException] with the given exception message.  */
  constructor(msg: String?) : super(msg) {}

  /** Constructs a new [StoreException] with the given exception.  */
  constructor(e: Exception?) : super(e) {}
}